### PR TITLE
Relax checks for extended capability to support new format

### PR DESCRIPTION
**Pull request by blueyed**
In the [20180331 release](https://github.com/mirror/ncurses/commit/74137fec), the format was slightly changed:

> 20180331
>         + improve terminfo write/read by modifying the fourth item of the
>           extended header to denote the number of valid strings in the extended
>           string table (prompted by a comment in unibilium's sources).

Since the number of valid string capabilities is not necessarily the
same as extstrslen, it's not possible to sanity check the total number
of items up front anymore.

Include a new test with an updated screen terminfo dump.

### CODE REVIEW

**Comment by blueyed** [comment]
Hmm, `make test` still passes with the new test, but the patch to the source removed?!
```diff
diff --git i/unibilium.c w/unibilium.c
index 8586b6f..64472de 100644
--- i/unibilium.c
+++ w/unibilium.c
@@ -339,6 +339,8 @@ unibi_term *unibi_from_mem(const char *p, size_t n) {
             extalllen += extnumlen;
             extalllen += extstrslen;

+            DEL_FAIL_IF(extofflen != extalllen + extstrslen, EINVAL, t);
+
             DEL_FAIL_IF(
                 n <
                 extboollen +
```

**Comment by blueyed** [comment]
To be clear: I'd expected the test to fail then.
Since it is not clear to me what the patch is about, at least the test should test for it I think.

**Comment by justinmk** [comment]
We need this stuff in the 0.4 release, so would be better to merge now and ask questions later :)

**Comment by blueyed** [comment]
Yeah :) - but it's not like we need to hurry it "just because", without waiting a few days at least.

**Comment by jamessan** [comment]
> Hmm, `make test` still passes with the new test, but the patch to the source removed?!

Hmm, guess I didn't choose a good terminfo entry to use for the test.

**Comment by blueyed** [comment]
@jamessan 
So where was the problem seen with then?
Would it happen with newer files from ncurses (development, after 6.1)?

**Comment by blueyed** [comment]
Ok.  I've used the screen terminfo from https://github.com/mauke/unibilium/issues/40, and when generating the test for it it fails without the patch to `unibilium.c` here:

> % TERMINFO_DIRS=$PWD tools/gen-static-test screen > t/static_screen.c
> unibi_from_term("screen"): No such file or directory

With this patch it works, and the output is the same.

So the problem with the test not failing is that the patch is needed in the first place to generate the test, which is then using `unibi_from_mem` directly - but with the result that `unibi_dump` produced.

Therefore I think the patch is good, but the test adds no benefit (at least with regard to the patch).

**Comment by blueyed** [comment]
Should be good now.
Without the patch the tests will fail now:

> t/new-extended-format-static_screen.t .. 1/541 Bailout called.  Further testing stopped:  Invalid argument

This ships `t/fixtures/s/screen` now (from https://github.com/mauke/unibilium/issues/40).
I guess this might be just an updated version, and we would see the issue with ncurses 6.2 anyway already then - but that's not clear to me.

File: t/new-extended-format-static_screen.c [hunk 1]
```diff
@@ -0,0 +1,731 @@
...
+        ok(11 < n_ext && unibi_get_ext_str(ut, 11) == NULL, "ext_str[11].value = null");
+        unibi_add_ext_str(dt, "xm", NULL);
+        ok(11 < n_ext && strcmp(unibi_get_ext_str_name(ut, 11), "xm") == 0, "ext_str[11].name = \"%s\"", "xm");
+    }
+
+    {
+        char buf1[4096];
+        char buf2[4096];
+        size_t r1 = unibi_dump(ut, buf1, sizeof buf1);
+        size_t r2 = unibi_dump(ut, buf2, sizeof buf2);
```
Comment (jamessan) [inline t/new-extended-format-static_screen.c]: Shouldn't this be `dt`, not `ut`?

Comment (blueyed) [reply t/new-extended-format-static_screen.c]: Yeah, thanks!
Fixed.

**Pull request merged**